src/library/scala/collection/immutable/LazyList.scala

```diff
@@ -989,32 +1041,27 @@ final class LazyList[+A] private(private[this] var lazyState: () => LazyList.Sta
   * @define coll lazy list
   * @define Coll `LazyList`
   */
-@SerialVersionUID(3L)
+@SerialVersionUID(4L)
 object LazyList extends SeqFactory[LazyList] {
-  // Eagerly evaluate cached empty instance
-  private[this] val _empty = newLL(State.Empty).force
 
-  private sealed trait State[+A] extends Serializable {
-    def head: A
-    def tail: LazyList[A]
-  }
+  // LazyListTest.countAlloc
+  // var k = 0
+  // def kount(): Unit = k += 1
 
-  private object State {
-    @SerialVersionUID(3L)
-    object Empty extends State[Nothing] {
-      def head: Nothing = throw new NoSuchElementException("head of empty lazy list")
-      def tail: LazyList[Nothing] = throw new UnsupportedOperationException("tail of empty lazy list")
-    }
+  private object Uninitialized extends Serializable
+  private object MidEvaluation
 
-    @SerialVersionUID(3L)
-    final class Cons[A](val head: A, val tail: LazyList[A]) extends State[A]
-  }
+  private val EmptyFun: () => Null = () => null
+  private val Empty: LazyList[Nothing] = new LazyList(EmptyFun)
+  // Eagerly evaluate cached empty instance. `force` is called after the `Empty` field is initialized because
+  // the `evaluated` and `isEmpty` methods read the `Empty` field, which would be `null` otherwise.
+  locally(Empty.force)
```

Comment:
I think we can use the eager constructor (or `eagerCons`) for this

```suggestion
  private val Empty: LazyList[Nothing] = new LazyList(null, null)
```